lib/liquid/tags/render.rb

```diff
@@ -44,14 +44,16 @@ def render_tag(context, output)
       context_variable_name = @alias_name || template_name.split('/').last
 
       render_partial_func = ->(var) {
-        inner_context = context.new_isolated_subcontext
-        inner_context.template_name = template_name
-        inner_context.partial = true
-        @attributes.each do |key, value|
-          inner_context[key] = context.evaluate(value)
+        partial.with_context_restrictions(context) do
```

Comment:
I feel like this is the role of `new_isolated_subcontext`. This is just a further restriction and think It should be handled in there.